### ISSUE TITLE
botocore.auth supports `ReadOnlyCredentials`

### ISSUE THREAD

**Issue by relsunkaev** [vemel/mypy_boto3_builder#118]
**Describe the bug**
All instances of `Credentials` use in `botocore.auth` (`SigV2Auth`, `SigV3Auth`, `SigV4Auth`, `HmacV1Auth`, etc.) are fully compatible with `ReadOnlyCredentials` since they only require the object to have `access_key`, `secret_key`, and `token` attributes. In fact, `Credentials.method` is never used within `botocore` and `Credentials.get_frozen_credentials()` is only used in `botocore.signers.RequestSigner.get_auth_instance()`, so the two are almost always compatible.

**To Reproduce**
1. Install 'boto3-stubs'
2.  Run mypy on the following code sample

```python
import boto3
from botocore.auth import SigV4Auth

session = boto3.Session()
credentials = session.get_credentials()
frozen_credentials = credentials.get_frozen_credentials()

SigV4Auth(frozen_credentials, "service_name", "us-east-1")
```
3. mypy output

```
error: Argument 1 to "SigV4Auth" has incompatible type "ReadOnlyCredentials"; expected "Credentials"
```

**Comment by vemel** [comment]
Thank you for the report!

The issue has been fixed in the latest `mypy-boto3-builder 6.3.1`. So, expect it to be included in the upcoming `botocore-stubs 1.23.33` release. Should be out in a couple of hours.

Let me know if this new release works as it should or if you still have any issues.

**Comment by relsunkaev** [comment]
Works perfectly, thanks!
